packages/manager-components/src/hooks/datagrid/useResourcesV6.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState, useEffect } from 'react';
+import isDate from 'lodash.isdate';
+import { IcebergFetchParamsV6, fetchIcebergV6 } from '@ovh-ux/manager-core-api';
+import { useQuery } from '@tanstack/react-query';
+import { ColumnSort } from '../../components';
+
+export interface ColumnDatagrid {
+  cell: (props: any) => React.JSX.Element;
+  id: string;
+  label: string;
+  type: string;
+}
+
+interface ResourcesV6Hook {
+  queryKey: string[];
+  columns?: ColumnDatagrid[];
+}
+
+export function dataType(a: any) {
+  if (Number.isInteger(a)) return 'number';
+  if (isDate(a)) return 'date';
+  return typeof a;
+}
+
+function sortColumn(type: string, a: any, b: any, desc: boolean) {
+  switch (type) {
+    case 'number':
+      return desc
+        ? parseFloat(a) - parseFloat(b)
+        : parseFloat(b) - parseFloat(a);
+    case 'date':
+      return desc
+        ? new Date(a).getTime() - new Date(b).getTime()
+        : new Date(b).getTime() - new Date(a).getTime();
+    case 'boolean':
+      return desc ? Number(a) - Number(b) : Number(b) - Number(a);
+    case 'string':
+      return desc
+        ? a
+            ?.trim()
+            .toLowerCase()
+            ?.localeCompare?.(b?.trim().toLowerCase())
+        : b
+            .trim()
+            ?.toString()
+            ?.toLowerCase()
+            ?.localeCompare?.(a?.trim()?.toLowerCase());
+    default:
+      return -1;
```

Comment:
**nitpick:** We can have sorting for string and default
```suggestion
    case 'string':
    default:
      return desc
        ? a
            ?.trim()
            .toLowerCase()
            ?.localeCompare?.(b?.trim().toLowerCase())
        : b
            .trim()
            ?.toString()
            ?.toLowerCase()
            ?.localeCompare?.(a?.trim()?.toLowerCase());
```

Reply:
I take an exemple with a column with 'number' type, if you have a 'null' inside your data response, its not a 'string', 'number' or 'date' so you need to have a default sorting that cover this case